### PR TITLE
Login logout menu tweaks

### PR DESCRIPTION
**Pull request by keithamus**
Supersedes https://github.com/economist-components/component-navigation/pull/32

See commit logs for more details.

See [here for a demo](https://keithamus.github.io/component-navigation/)

Fixes: [TEC-541](https://jira.economist.com/browse/TEC-541), [TEC-542](https://jira.economist.com/browse/TEC-542), [TEC-543](https://jira.economist.com/browse/TEC-543)

### CODE REVIEW

File: index.es6 [hunk 1]
```diff
@@ -35,45 +35,81 @@ export default class Navigation extends React.Component {
...
+                Newsletters
+              </a>
+            </li>
+            <li className="navigation__user-menu-linklist-item">
+              <a className="navigation__user-menu-linklist-link" href="/users/guest-olejses/comments">
+                My comments
+              </a>
+            </li>
+            <li className="navigation__user-menu-linklist-item">
+              <a className="navigation__user-menu-linklist-link" href="/logout?destination=">
```
Comment (umbobabo) [inline index.es6]: Is destination expected to be empty?

File: index.es6 [hunk 2]
```diff
@@ -35,45 +35,81 @@ export default class Navigation extends React.Component {
...
+                Buy a gift subscription
+              </a>
+            </li>
+            <li className="navigation__user-menu-linklist-item">
+              <a className="navigation__user-menu-linklist-link" href="/users/guest-olejses/newsletters">
+                Newsletters
+              </a>
+            </li>
+            <li className="navigation__user-menu-linklist-item">
+              <a className="navigation__user-menu-linklist-link" href="/users/guest-olejses/comments">
```
Comment (umbobabo) [inline index.es6]: Is this implementation a prototype?

File: index.es6 (hunk 1 above)
Comment (keithamus) [inline index.es6]: I suppose so. The given spec had this as the link. I could adjust the destination but the spec made no mention of it.

File: index.es6 [hunk 3]
```diff
@@ -31,49 +34,136 @@ export default class Navigation extends React.Component {
...
+                Manage my subscription
+              </a>
+            </li>
+            <li className="navigation__user-menu-linklist-item">
+              <a className="navigation__user-menu-linklist-link" href="/subscriptions/renew">
+                Renew my subscription
+              </a>
+            </li>
+            <li className="navigation__user-menu-linklist-item">
+              <a className="navigation__user-menu-linklist-link" href="https://subscriptions.economist.com">
```
Comment (umbobabo) [inline index.es6]: Probably this should point to the correct tab:

https://subscriptions.economist.com/GLB/SPLASH/ECOM/UK/?absplitsrc=/GLB/ECOM&abrnd=56c47e2601913&country=GB&sub_type=donation ?

File: index.es6 [hunk 4]
```diff
@@ -31,49 +34,136 @@ export default class Navigation extends React.Component {
...
+      return (
+        <Balloon showOnHover trigger={userMenuBalloonTrigger}>
+          <ul className="navigation__user-menu-linklist">
+            <li className="navigation__user-menu-linklist-item">
+              <a className="navigation__user-menu-linklist-link" href="/user">
+                My account
+              </a>
+            </li>
+            <li className="navigation__user-menu-linklist-item">
+              <a className="navigation__user-menu-linklist-link" href="https://subscriptions.economist.com">
```
Comment (umbobabo) [inline index.es6]: Lot of duplications in the links hrefs.
Shouldn't we create CONST for the same links to be less prone error in the updates?

File: index.es6 [hunk 5]
```diff
@@ -31,49 +34,136 @@ export default class Navigation extends React.Component {
...
-          </Button>
-          <span className="navigation__user-menu-register">
-            New to <span className="navigation__user-menu-the-economist-name">The Economist</span>?
-            <a
-              className="navigation__user-menu-register-link"
-              href={registerUrl}
-            >Register now</a>
-          </span>
-        </div>
+        <Button shadow href={buttonUrl} className="navigation__user-menu-log-in-button">
```
Comment (umbobabo) [inline index.es6]: Can you please remove "shadow" because Sue don't want the shadow anymore.

File: user-menu.css [hunk 6]
```diff
@@ -20,12 +20,9 @@
   align-items: center;
 }
 
-.navigation__user-menu-link--logout {
-  color: var(--navigation-user-icon-color-logout, var(--color-boston));
-}
-
-.navigation__user-menu-link--login {
-  font-size: var(--search__search-label__font-size, var(--navigation-text-size));
+.navigation__user-menu-link--logout, .navigation__user-menu-link--login {
```
Comment (umbobabo) [inline user-menu.css]: I prefer a line brake after the comma.

File: user-menu.css [hunk 7]
```diff
@@ -20,12 +20,9 @@
...
+  color: var(--navigation-user-menu-color, white);
```
Comment (umbobabo) [inline user-menu.css]: Instead 'white' I think you should use the var from the white from the component-palette:

https://github.com/economist-components/component-palette/blob/master/index.css#L12

File: index.es6 (hunk 3 above)
Comment (keithamus) [inline index.es6]: The spec gave explicitly the link I put the in the code. I'd rather not change it without taking it up with Jay first.

File: index.es6 (hunk 4 above)
Comment (keithamus) [inline index.es6]: I don't see much of a benefit to this. If they later change to different things then we'll have to more heavily refactor the code. I considering making this more DRY but at the end of the day thought it'd just be easier to have it as simple plain html.

File: index.es6 (hunk 3 above)
Comment (umbobabo) [inline index.es6]: Specs are not so accurate unfortunately. Just ping him please. Anyway this is not blocking.

Comment (keithamus) [inline index.es6]: I have asked Jay but he hasn't got back. Could we merge this work now and change later as it wont be breaking, and is a quick fix. Thanks :smile:

**Pull request merged**